### DISCUSSION TITLE
Breaking changes in Z Open Editor v5.0.0

### DISCUSSION THREAD

**Discussion by phaumer** [IBM/zopeneditor-about#435] [Announcements]
As we did previously for major releases (e.g. [Z Open Editor 3.0.0](https://github.com/IBM/zopeneditor-about/discussions/284)) we want to warn you about a major release update that has some breaking changes that require updates to your development machines. We cannot announce the date for this release just yet, but we want to give our users enough time to prepare.

With Z Open Editor v5.0.0 there will be two major changes:

- Support for a Java 11 runtime will be removed: Users running Z Open Editor on their local systems will require a Java 17 or newer runtime. The editor will not work with Java 11 anymore. We recommend [IBM's Semeru Runtime or JDK](https://developer.ibm.com/languages/java/semeru-runtimes/downloads/). Distributions from Oracle and OpenJDK will also work. Switching to Java 17 is long overdue for us and it will allow us to finally adopt newer versions of LSP4J and work with some of the exciting features that it provides.
- Zowe v3 will be required: When installing Z Open Editor v5 it will downloads and installs Zowe Explorer v3. Zowe Explorer v2 will not work anymore. For IBM RSE API Plugin for Zowe CLI the requirement will be Zowe CLI v8. [Zowe v3 is announced](https://docs.zowe.org/stable/whats-new/zowe-announcements/) for September 30, 2024. See the [zowe.org/vnext](https://www.zowe.org/vnext) page with all of the changes announced for this Zowe release. Note, that users need to wait to upgrade to Zowe Explorer v3 until Z Open Editor 5.0 is available. We recommend disabling Zowe Explorer's auto update shortly before the Zowe v3 release. See the [VS Code documentation](https://code.visualstudio.com/docs/editor/extension-marketplace#_extension-autoupdate) for how to easily do this in the VS Code Extensions view.

**Comment by FALLAI-Denis** [comment]
Hi @phaumer,

Does Zowe CLI (V3 or other) need to be physically installed on the workstation for Z Open Editor / Zowe Explorer to work? (This is the case on our workstations, but for direct use and not necessarily for ZOE / ZE).
A priori ZOE and ZE already embed the Zowe SDK, and just need a declaration of Zowe profiles (or RSE profiles).
In our operating context, the ZOWE (or RSE) profiles (team profiles in project configuration files) are embedded in the workspace (included in the Git repository used to create the workspace).

Is the structure of the profile files and associated schemas changed by the switch to Zowe V3? If so, ZOE or ZE must support automatic V2 / V3 migration to avoid reproducing the problem known when changing the structure of the zapp.yaml file.

Thanks.

**Comment by phaumer** [comment]
@FALLAI-Denis no, Zowe Explorer v3 does not require an installation of Zowe CLI. Also the format of the Zowe CLI profiles will not have any breaking changes in v3 that would require a migration.

The only thing Z Open Editor v5 requires is that Zowe Explorer v3 is installed as the breaking changes are only for use extenders so that we cannot also support running Zowe Explorer v2. Vice versa, Z Open Editor v4 will not work with Zowe Explorer v3. 

Here is a detailed list of the planned Zowe Explorer changes: <https://github.com/zowe/zowe-explorer-vscode/wiki/v3-Changes-for-Users-and-Extenders>

**Comment by SMoRG75** [comment]
Hi, will there be a preview version of Z Open Editor 5.0.0, in the near future, similar to Zowe v3 preview, so our own extension can be made ready for 5.0.0.

Thanks.

**Comment by phaumer** [comment]
@SMoRG75 and everyone: We published an early alpha-level pre-release version here: <https://github.com/IBM/zopeneditor-about/releases/tag/5.0.0-alpha1>

**Comment by FALLAI-Denis** [comment]
Hi,

Zowe Explorer version 3.0.0 has been released to the Marketplace, and installs automatically, which then causes IBM Z Open Editor 4.5.x to malfunction...
Zowe Explorer needs to be downgraded to version 2.18.0.
See:
* #448

**Comment by phaumer** [comment]
Version 5 of Z Open Editor is now available on the VS Code Marketplace, Open-VSX, and our Mainframe Download site.

**Comment by TommyTechh** [comment]
Hi,
Do we have any news on when a version where copybooks are handled in-memory is available or open for testing?
We want to convert our solutions to use zowe v3 and ibm z open editor v5+ but we're waiting until everything is handled in the in-memory FileSystemProvider.

Thanks in advance

**Comment by phaumer** [comment]
Next release. I cannot say the exact date, but not too long.

**Comment by phaumer** [comment]
@TommyTechh we now released v5.1.0 which fully adopts the Zowe Explorer virtual file system. See [here](https://ibm.github.io/zopeneditor-about/Blog/new-content.html#ibm-z-open-editor-and-ibm-rse-api-plug-in-for-zowe-cli-v5-1-0-released) for more details.

**Reply by TommyTechh** [reply]
Thanks for letting me know, we've put the migration of our tools into next sprint!
